src/mpid/ch4/shm/posix/eager/iqueue/iqueue_send.h

```diff
@@ -19,116 +19,75 @@ MPL_STATIC_INLINE_PREFIX size_t MPIDI_POSIX_eager_buf_limit(void)
     return MPIR_CVAR_CH4_SHM_POSIX_IQUEUE_CELL_SIZE;
 }
 
-/* This function attempts to send the next chunk of a message via the queue. If no cells are
- * available, this function will return and the caller is expected to queue the message for later
- * and retry.
- *
- * grank   - The global rank (the rank in MPI_COMM_WORLD) of the receiving process.
- * msg_hdr - The header of the message to be sent. This can be NULL if there is no header to be sent
- *           (such as if the header was sent in a previous chunk).
- * iov     - The array of iovec entries to be sent.
- * iov_num - The number of entries in the iovec array.
+/* This function attempts to allocate a cell and return the address and size of the payload portion
+ * of the cell.
  */
-MPL_STATIC_INLINE_PREFIX int
-MPIDI_POSIX_eager_send(int grank,
-                       MPIDI_POSIX_am_header_t ** msg_hdr, struct iovec **iov, size_t * iov_num)
+MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_eager_get_buf(void **eager_buf, size_t * eager_buf_sz)
 {
+    int mpi_errno = MPI_SUCCESS;
+    MPIDI_POSIX_eager_iqueue_cell_t *cell = NULL;
     MPIDI_POSIX_eager_iqueue_transport_t *transport;
-    MPIDI_POSIX_eager_iqueue_cell_t *cell;
-    MPIDU_genq_shmem_queue_t terminal;
-    size_t i, iov_done, capacity, available;
-    char *payload;
-    int ret = MPIDI_POSIX_OK;
+    void *payload = NULL;
 
-    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_POSIX_EAGER_SEND);
-    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_POSIX_EAGER_SEND);
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_POSIX_EAGER_SEND_SHORT);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_POSIX_EAGER_SEND_SHORT);
 
     /* Get the transport object that holds all of the global variables. */
     transport = MPIDI_POSIX_eager_iqueue_get_transport();
 
-    /* TODO: This function can only send one cell at a time. For a large
```

Comment:
@yfguo the reason I am interested is this comment, which you may have addressed? I am wondering if this has positive performance implications.